networkx/utils/backends.py

```diff
@@ -53,142 +53,161 @@
-
--   If a backend only partially implements some algorithms, it can define
-    a ``can_run(name, args, kwargs)`` function that returns True or False
-    indicating whether it can run the algorithm with the given arguments.
-    It may also return a string indicating why the algorithm can't be run;
-    this string may be used in the future to give helpful info to the user.
-
--   A backend may also define ``should_run(name, args, kwargs)`` that is similar
-    to ``can_run``, but answers whether the backend *should* be run (converting
-    if necessary). Like ``can_run``, it receives the original arguments so it
-    can decide whether it should be run by inspecting the arguments. ``can_run``
-    runs before ``should_run``, so ``should_run`` may assume ``can_run`` is True.
-    If not implemented by the backend, ``can_run`` and ``should_run`` are
-    assumed to always return True if the backend implements the algorithm.
-
--   A special ``on_start_tests(items)`` function may be defined by the backend.
-    It will be called with the list of NetworkX tests discovered. Each item
-    is a test object that can be marked as xfail if the backend does not support
-    the test using ``item.add_marker(pytest.mark.xfail(reason=...))``.
-
--   A backend graph instance may have a ``G.__networkx_cache__`` dict to enable
-    caching, and care should be taken to clear the cache when appropriate.
+2. Running Tests:
+    You can invoke NetworkX tests for your custom backend with the following commands::
+
+        NETWORKX_TEST_BACKEND=<backend_name>
+        NETWORKX_FALLBACK_TO_NX=True # or False
+        pytest --pyargs networkx
+
+How tests are run?
+~~~~~~~~~~~~~~~~~~
+
+1. While dispatching to the backend implementation the ``_convert_and_call`` function
+   is used and while testing the ``_convert_and_call_for_tests`` function is used.
+   Other than testing it also checks for functions that return numpy scalars, and
+   for functions that return graphs it runs the backend implementation and the
+   networkx implementation and then converts the backend graph into a NetworkX graph
+   and then compares them, and returns the networkx graph. This can be regarded as
+   (pragmatic) technical debt. We may replace these checks in the future.
+
+2. Conversions while running tests :
+
+       - Convert NetworkX graphs using ``<your_dispatcher_class>.convert_from_nx(G, ...)`` into
+         the backend graph.
+       - Pass the backend graph objects to the backend implementation of the algorithm.
+       - Convert the result back to a form expected by NetworkX tests using 
+         ``<your_dispatcher_class>.convert_to_nx(result, ...)``.
+       - For nx-loopback, the graph is copied using the dispatchable metadata
+
+3. Dispatchable algorithms that are not implemented by the backend
+   will cause a ``pytest.xfail``, giving some indication that not all
+   tests are running, while avoiding causing an explicit failure.
```

Comment:
This runs `xfail` only if `fallback_to_nx` is False (controlled by the `NETWORKX_FALLBACK_TO_NX` env variable). See logic here:
https://github.com/networkx/networkx/blob/fd27fb0d9a579e6dd2e58f1f7762456141304ebb/networkx/utils/backends.py#L1189-L1198